Scan API and get_account_transfers (#1054)
tigerbeetle/tigerbeetle: docs/reference/operations/get_account_transfers.md

# `get_account_transfers`

> Fetch [`Transfer`](../transfers.md)s from a given [`Account`](../accounts.md).

Reviewer: Should this be "to or from" instead of "from"? Or maybe "involving".. (trying to imply that either direction works).

## Event

> An struct containing the query parameters: 

Reviewer: ```suggestion
A struct containing the query parameters: 
```

Also since this is the user-level documentation I'm not sure that we want to refer to it as a "struct" -- that is an internal implementation detail.

### `account_id`

> The unique identifier of the account which the transfers will be retrieved.

Reviewer: ```suggestion
The unique identifier of the account for which the transfers will be retrieved.
```

# `get_account_transfers`

Fetch [`Transfer`](../transfers.md)s involving a given [`Account`](../accounts.md).

## Event

The query filter: 

### `account_id`

The unique identifier of the account for which the transfers will be retrieved.

Constraints:

* Type is 128-bit unsigned integer (16 bytes)
* Must not be zero or `2^128 - 1`

### `timestamp`

> The [`timestamp`](../transfers.md#timestamp) from which results will be returned, exclusive range

Reviewer: ```suggestion
The [`Transfer.timestamp`](../transfers.md#timestamp) from which results will be returned, exclusive range
```

Since there is also a field named `timestamp` in this file, hint that we are referring to something a little different (albeit related) here.

> applied as greater than `>` or less than `<` if in [reversed](#flagsreversed) chronological order.

Reviewer: Nit, but let's make this a little clearer:

```suggestion
applied as:
- greater than `>` (when [reversed](#flagsreversed) is `false`)
- less than `<` (when [reversed](#flagsreversed) is `true`)
```

Optional, use zero for no filter by timestamp.

Constraints:

* Type is 64-bit unsigned integer (8 bytes)
* Must not be `2^64 - 1`

### `limit`

The maximum number of [`Transfer`](../transfers.md)s that can be returned by this query.

Limited by the [maximum message size](../../design/client-requests.md#batching-events).

Constraints:

* Type is 32-bit unsigned integer (4 bytes)
* Must not be zero

### `flags`

A bitfield that specifies querying behavior.

Constraints:

* Type is 32-bit unsigned integer (4 bytes)

#### `flags.debits`

Whether or not to include [`Transfer`](../transfers.md)s where the field [`debit_account_id`](../transfers.md#debit_account_id)
matches the parameter [`account_id`](#account_id).

#### `flags.credits`

Whether or not to include [`Transfer`](../transfers.md)s where the field [`credit_account_id`](../transfers.md#credit_account_id)
matches the parameter [`account_id`](#account_id).

#### `flags.order`

Whether to sort results by timestamp in ascending (flag _unset_) or descending (flag _set_) order.

## Result

- If the transfers exist, return an array of [`Transfer`](../transfers.md)s.
- If the transfer does not exist, return nothing.

> - If any constraint is violated, it returns nothing.

Reviewer: ```suggestion
- If any matching transfers exist, return an array of [`Transfer`](../transfers.md)s.
- If no matching transfers exist, return nothing.
- If any constraint is violated, return nothing.
```
